x-pack/plugins/canvas/public/app.js

```diff
@@ -23,5 +24,10 @@ import 'uiExports/fieldFormats';
 import './lib/load_expression_types';
 import './lib/load_transitions';
 
-// load the application
-chrome.setRootController('canvas', CanvasRootController);
+// Redirect to Kibana home if canvas shouldn't be visible
+if (!uiCapabilities.canvas.show) {
```

Comment:
This is proving problematic for the Karma test runner. The runner creates `uiCapabilities` via the OSS `tests_bundle` / [`tests_entry_template`](https://github.com/elastic/kibana/blob/05cd396c848e8b27c7c52782ed6ea7d20da4447f/src/legacy/core_plugins/tests_bundle/tests_entry_template.js).

Up until now, we've gotten away with hard coding the list of capabilities there, but now we've gotten to the point where Non-OSS plugins need to read this list, and they should expect that their capabilities are available.

The easy way out is to include the x-pack plugin capabilities inside `tests_entry_template`, but that's not the right way to do it. I'm not sure how else to do this though, if the client-side is going to be responsible for this redirect. I suppose an alternative is to only react if `uiCapabilities.canvas` is defined, but I'd rather not take that approach, as it's something that shouldn't happen in real life, and would only serve to confuse readers.

Do you have any ideas @kobelb?